### PR TITLE
refactor(cohort): split out cohortLogic into cohortLogic and cohortEditLogic

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

Cohort breadcrumbs didn't work nicely with scene logic so it was disabled on purpose. 

## Changes

- Split out cohortLogic into a separate cohortLogic and cohortEditLogic. This decouples logic that is attached to the cohort detail scene from the form logic. This allows form logic to be unmounted/mounted on demand. 
- Replace buggy BindLogic with props passing

Screen Shot 2022-05-17 at 2 13 10 PM

Breadcrumbs are back

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

cohortEditLogic tests pass

### CODE REVIEW

File: frontend/src/scenes/cohorts/CohortFilters/CohortCriteriaRowBuilder.stories.tsx [hunk 1]
```diff
@@ -23,9 +24,11 @@ export function _CohortCriteriaRowBuilder(props: CohortCriteriaRowBuilderProps):
     useMountedLogic(actionsModel)
     useMountedLogic(cohortsModel)
     useMountedLogic(cohortLogic({ id: 1 }))
```
Comment (rcmarron) [inline frontend/src/scenes/cohorts/CohortFilters/CohortCriteriaRowBuilder.stories.tsx:26]: Nit: I think we can remove `cohortLogic` from here

Comment (rcmarron) [inline frontend/src/scenes/cohorts/CohortFilters/CohortCriteriaRowBuilder.stories.tsx:26]: Nit: I don't think we need this logic anymore

File: frontend/src/scenes/cohorts/CohortFilters/CohortNumberField.stories.tsx [hunk 2]
```diff
@@ -13,6 +14,7 @@ export default {
 
 const Template: ComponentStory<typeof CohortNumberField> = (props: CohortNumberFieldProps) => {
     useMountedLogic(cohortLogic({ id: 1 }))
```
Comment (rcmarron) [inline frontend/src/scenes/cohorts/CohortFilters/CohortNumberField.stories.tsx:16]: Nit: same for this logic

File: frontend/src/scenes/cohorts/cohortEditLogic.ts [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+            },
+        ],
+    })),
+
+    selectors({
+        newCohortFiltersEnabled: [
+            () => [featureFlagLogic.selectors.featureFlags],
+            (featureFlags) => !!featureFlags[FEATURE_FLAGS.COHORT_FILTERS],
+        ],
+        breadcrumbs: [
```
Comment (rcmarron) [inline frontend/src/scenes/cohorts/cohortEditLogic.ts:256]: Can we get rid of this breadcrumbs selector?

**Review by rcmarron — COMMENTED** [review]
A couple nits below, but outside of that, I'm not sure I see why we need the refactor into 2 logics to solve the breadcrumb issue. It seems like it leaves us with a pretty minimal `cohortLogic` that just does the breadcrumbs, and then everything else is in `cohortEditLogic` including stuff that isn't really related to editing a cohort (e.g. `loadCohort` or `checkIfFinishedCalculating`)

Could we also solve the bug by just adding `logic: cohortLogic` in the scene export?

**Comment by alexkim205** [comment]
Hey @rcmarron thanks for the review! 

Unfortunately adding `cohortLogic` to `sceneExport` introduces the problem where cohortLogic will stay mounted across the cohorts list and cohort detail pages. This creates bugs like cohort form data persisting across different cohort detail pages (you can reproduce this by setting cohortLogic in scene export, creating and saving a new cohort, returning to cohorts list, and creating a new cohort). Keeping the cohort page logic and the cohort edit logic in a single logic makes it impossible to isolate the mounting and unmounting of logic that should be reset in different contexts. Let me know if this makes any sense or if I can clarify the reasoning.

**Review by rcmarron — APPROVED** [review]
🚢

Alex and I had a quick call on the need for this logic split, and we're on the same page for why we need it for the breadcrumbs to behave correctly.

**Pull request merged**